tests/test_forms.py

```diff
@@ -33,3 +37,330 @@ class Meta:
+        self.assertEqual(
+            list(form.fields.keys()),
+            [
+                "title_fr",
+                "title",
+                "start_date",
+                "default_language",
+                "header_fr",
+                "header",
+                "end_date",
+            ],
+        )
+
+    def test_fields_defined_with_exclude_options(self):
+        """Test fields and their order defined with Meta 'exclude' option."""
+        form = ExcludeForm()
+        self.assertEqual(form.language_codes, ["en", "de", "fr"])
+        self.assertEqual(
+            list(form.fields.keys()), ["start_date", "title", "title_de", "title_fr", "end_date"]
+        )
+
+    def test_fields_defined_with_exclude_option_tuple(self):
+        """Test that form works correctly even if fields is defined in tuple format."""
+
+        class TupleForm(TranslationModelForm):
+            class Meta:
+                model = Challenge
+                exclude = ("default_language",)
+
+        form = TupleForm()
+        self.assertEqual(list(form.fields.keys()), ["title", "header"])
+
+    def test_fields_with_languages_kwarg_with_exclude_option(self):
+        """Test fields and their order with parameter override in form with 'exclude' option."""
+        form = ExcludeForm(languages=["de", "fallback"], fallback_language="nl")
+        self.assertEqual(form.language_codes, ["de", "nl"])
+        self.assertEqual(
+            list(form.fields.keys()), ["start_date", "title_de", "title_nl", "end_date"]
+        )
+
+    def test_fields_with_model_instance_fallback_with_exclude_options(self):
+        """Test fields and their order with model instance fallback override in form with 'exclude' option."""
+        challenge = Challenge.objects.create(default_language="nl")
+        form = ExcludeForm(instance=challenge)
+        self.assertEqual(form.language_codes, ["en", "de", "fr"])
+        self.assertEqual(
+            list(form.fields.keys()), ["start_date", "title", "title_de", "title_fr", "end_date"]
+        )
+
+    def test_fields_with_fallback_language_kwarg_with_exclude_option(self):
+        """Test fields and their order with parameter override of model instance fallback in form with 'exclude'."""
+        challenge = Challenge(default_language="de")
+        form = ExcludeForm(instance=challenge, fallback_language="nl")
+        self.assertEqual(form.language_codes, ["en", "de", "nl"])
+        self.assertEqual(
+            list(form.fields.keys()), ["start_date", "title", "title_de", "title_nl", "end_date"]
+        )
+
+    def test_setting_of_field_properties(self):
+        """Test that fields are set with the correct properties."""
+        with self.subTest("Browser (fallback) language"):
+            title_field = Form().fields["title"]
+            self.assertEqual(title_field.label, "Title (EN, default language)")
+            self.assertEqual(title_field.required, True)
```

Comment:
Ik denk dat het goed is dat we voor `assertEqual(x, True|False)` wel `assertTrue(x)` en `assertFalse(x)` gebruiken, nu is een beetje willekeurig.
```suggestion
            self.assertTrue(title_field.required)
```